### ISSUE TITLE
API documentation

### ISSUE THREAD

**Issue by alcalyn** [pakastin/deck-of-cards#38]
Hi, searching for an HTML5 playing card module I could use to develop card games, this one is the best I've found.

The [Simple example of usage](http://jsfiddle.net/x0gjood1/) is cool and learns a lot of how I could use it.

But is there an API documentation that would explain many things like:
- Make a card as draggable
- Reveal or hide programmatically a card
- Prevent a card to be revealed
- Creating a fan with a subset of a deck
- How to use shuffle or sort modules
- Adding or removing a card in a deck...

**Comment by pakastin** [comment]
I'm recreating the library, new version will support those..

**Comment by alcalyn** [comment]
Well ok, I thought the work you described here https://github.com/pakastin/deck-of-cards/issues/28 was already merged.

**Comment by pakastin** [comment]
This wasn't actually initially meant to be an API, but just an experiment (actually the first version was made with Flash over 10 years ago and then with Angular + TweenMax)..

- http://pakastin.fi/flash/kortit.html
- https://www.chromeexperiments.com/experiment/html5-deck-of-cards

**Comment by pakastin** [comment]
Sorry no, it's still work in progress..

**Comment by alcalyn** [comment]
I tested it more deeply, I could make some good things easily, like

``` js
deck.cards[0].enableDragging()
deck.cards[0].setSide('front')
deck.cards[0].disableFlipping()
deck.cards[0].shuffle()
deck.cards[0].sort()

var deck2 = new Deck();
deck2.cards.push(deck.pop());
```

Method names are well intuitive, and that answers at many of my use cases.

**Comment by pakastin** [comment]
Sounds legit! I haven't use this for a while myself, sorry I wasn't much help 🙂

**Comment by alcalyn** [comment]
nope, i'll submit a PR with that I could done.

Just a thing, do you remember how to update the deck cards view ? When I `deck.cards.pop()` a card, the view does not update.

**Comment by pakastin** [comment]
`card.unmount()` should do the trick ;)

https://github.com/pakastin/deck-of-cards/blob/master/lib/card.js#L218-L222

**Comment by alcalyn** [comment]
Started a documentation: https://github.com/pakastin/deck-of-cards/pull/39

**Comment by alcalyn** [comment]
Merged in #39 .
